File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        avail = 0
    
        for i in range(len(flowerbed)):
            if flowerbed[i] == 0 and (i == 0 or flowerbed[i - 1] != 1) and (i == len(flowerbed) - 1 or flowerbed[i + 1] != 1):
                avail += 1
                flowerbed[i] = 1

    
        return avail >= n
    

File: 0605-can-place-flowers/0605-can-place-flowers.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function(flowerbed, n) {
    let avail = 0
    
    for(let i = 0; i < flowerbed.length; i++){
        if((flowerbed[i] === 0 && flowerbed[i - 1] !== 1 && flowerbed[i + 1] === 0) || (flowerbed[i] === 0 && flowerbed[i - 1] !== 1 && flowerbed[i + 1] === undefined)){
            avail += 1
            flowerbed[i] = 1
        }
    }
    
    
    return avail >= n
};